hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/HoodieMergeHelper.java

```diff
@@ -122,12 +121,10 @@ public void runMerge(HoodieTable<?, ?, ?, ?> table,
             baseFileReader,
             HoodieFileReaderFactory.getReaderFactory(recordType).getFileReader(bootstrapFileConfig, bootstrapFilePath),
             mergeHandle.getPartitionFields(),
-            mergeHandle.getPartitionValues());
+            mergeHandle.getPartitionValues(),
+            table.getConfig().isBootstrapPartitionColumnTypeInferenceEnabled());
         recordSchema = mergeHandle.getWriterSchemaWithMetaFields();
-        recordIterator = new ClosableMergingIterator<>(
-            baseFileRecordIterator,
-            (ClosableIterator<HoodieRecord>) bootstrapFileReader.getRecordIterator(recordSchema),
-            (left, right) -> left.joinWith(right, recordSchema));
+        recordIterator = (ClosableIterator<HoodieRecord>) bootstrapFileReader.getRecordIterator(recordSchema);
```

Comment:
Not sure why the closableMergingIterator was still being used. We already do the merging once